assets/js/src/nps_poll.jsx

```diff
@@ -70,29 +55,35 @@ class NpsPoll extends React.Component {
       events: {
         submit: (response) => {
           if (response.rating >= 9 && response.completed) {
-            this.showReviewRequestModal();
+            showReviewRequestModal();
           }
         },
       },
     });
   }
 
-  render() {
-    if (!window.mailpoet_display_nps_poll) {
-      return null;
-    }
-    if (window.satismeter) {
-      setImmediate(this.displayPoll);
-    } else {
-      const s = document.createElement('script');
-      s.type = 'text/javascript';
-      s.src = 'https://app.satismeter.com/satismeter.js';
-      s.onload = () => this.displayPoll();
-      document.getElementsByTagName('body')[0].appendChild(s);
+  function displayPoll() {
+    if (!pollShown) {
+      setPollShown(true);
+      getTrackingData().then(data => callSatismeter(data.data));
     }
+  }
+
+  if (!window.mailpoet_display_nps_poll) {
     return null;
   }
-}
+
+  if (window.satismeter) {
```

Comment:
Why aren't we using `satismeter` as a module? like using this https://www.npmjs.com/package/satismeter-loader

Reply:
Good find. Originally we were loading the satismeter code on every admin page so, when I was adding new properties to satismeter I've [refactored so that it is loaded only when needed](https://github.com/mailpoet/mailpoet/pull/1876/commits/d11aaa41fb306e1c8611a195ab4280f25e1598ea). I did not look for such a module. The module hasn't been updated for 2 years but all it does is that it loads the satismeter script so I think we can use it. Should I add a commit or rather create a task for this?

Reply:
I was quick fix so I added it